tensorboard/webapp/deeplink/hash.ts

```diff
@@ -23,29 +22,24 @@ const TAB = '__tab__';
 
 @Injectable()
 export class HashDeepLinker implements DeepLinkerInterface {
-  private readonly tfStorage: TfStorageElement;
-
   constructor() {
-    this.tfStorage = document.createElement('tf-storage');
-    const tfGlobals = document.createElement('tf-globals');
-
     // Note: `migrateLegacyURLScheme()` must be called before `setUseHash`, so
     // that tfStorage reads from the actual URL, not the fake hash for tests
     // only.
-    tfGlobals.tf_globals.setUseHash(true);
-    this.tfStorage.tf_storage.migrateLegacyURLScheme();
+    window['tensorboard']['tf_storage']['migrateLegacyURLScheme']?.();
```

Comment:
If you properly declared the API interface for the Angular binary then shouldn't this be tensorboard.tf_storage.migrateLegacyURLScheme?

It should be unnecessary to access the API like this (and I don't think it'll impress the readability team).

Reply:
Spoke offline. Riley to look into how to fix this, possibly using typings, possibly using some internal examples of interactions with thirdparty js binaries as motivation.